src/aiida_workgraph/engine/task_manager.py

```diff
@@ -250,7 +248,8 @@ def run_tasks(self, names: List[str], continue_workgraph: bool = True) -> None:
             self.process.report(
                 f"Run task: {name}, type: {task['metadata']['node_type']}"
             )
-            executor, _ = get_executor(task["executor"])
+            executor = self.process.node.task_executors[name]
+            executor = NodeExecutor(**executor).executor if executor else None
```

Comment:
```suggestion
            executor_dict = self.process.node.task_executors[name]
            executor = NodeExecutor(**executor_dict).executor if executor_dict else None
```
Maybe? Otherwise the second line gives me a brain aneurysm with all the `executor` :D